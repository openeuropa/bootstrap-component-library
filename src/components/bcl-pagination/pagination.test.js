import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-data-pagination/data";

const template = "@oe-bcl/bcl-pagination/pagination.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);
const sizes = ["sm", "lg"];
const alignments = ["end", "center"];

describe("OE - Pagination", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  sizes.forEach((size) => {
    test(`${size} renders correctly`, () => {
      expect.assertions(1);

      return expect(render({ ...demoData, size })).resolves.toMatchSnapshot();
    });
  });

  alignments.forEach((alignment) => {
    test(`${alignment} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, alignment })
      ).resolves.toMatchSnapshot();
    });
  });

  test(`renders correctly as glossary`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, variant: "glossary" })
    ).resolves.toMatchSnapshot();
  });
});
