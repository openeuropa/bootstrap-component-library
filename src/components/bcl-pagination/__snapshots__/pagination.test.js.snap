// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Pagination center renders correctly 1`] = `
<jest>
  <nav
    aria-label="Test aria label"
    role="navigation"
  >
    <ul
      class="pagination justify-content-center"
    >
      <li
        class="page-item"
      >
        <a
          aria-label="First page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-left"
            />
          </svg>
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Previous page"
          class="page-link"
          href="/example.html"
        >
          Previous
        </a>
      </li>
      <li
        class="page-item disabled"
      >
        <a
          aria-disabled="true"
          class="page-link"
          href="/example.html"
          tabindex="-1"
        >
          1
        </a>
      </li>
      <li
        class="page-item active"
      >
        <a
          aria-current="page"
          class="page-link"
          href="/example.html"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="/example.html"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Next page"
          class="page-link"
          href="/example.html"
        >
          Next
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Last page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-right"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</jest>
`;

exports[`OE - Pagination end renders correctly 1`] = `
<jest>
  <nav
    aria-label="Test aria label"
    role="navigation"
  >
    <ul
      class="pagination justify-content-end"
    >
      <li
        class="page-item"
      >
        <a
          aria-label="First page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-left"
            />
          </svg>
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Previous page"
          class="page-link"
          href="/example.html"
        >
          Previous
        </a>
      </li>
      <li
        class="page-item disabled"
      >
        <a
          aria-disabled="true"
          class="page-link"
          href="/example.html"
          tabindex="-1"
        >
          1
        </a>
      </li>
      <li
        class="page-item active"
      >
        <a
          aria-current="page"
          class="page-link"
          href="/example.html"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="/example.html"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Next page"
          class="page-link"
          href="/example.html"
        >
          Next
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Last page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-right"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</jest>
`;

exports[`OE - Pagination lg renders correctly 1`] = `
<jest>
  <nav
    aria-label="Test aria label"
    role="navigation"
  >
    <ul
      class="pagination pagination-lg"
    >
      <li
        class="page-item"
      >
        <a
          aria-label="First page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-left"
            />
          </svg>
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Previous page"
          class="page-link"
          href="/example.html"
        >
          Previous
        </a>
      </li>
      <li
        class="page-item disabled"
      >
        <a
          aria-disabled="true"
          class="page-link"
          href="/example.html"
          tabindex="-1"
        >
          1
        </a>
      </li>
      <li
        class="page-item active"
      >
        <a
          aria-current="page"
          class="page-link"
          href="/example.html"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="/example.html"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Next page"
          class="page-link"
          href="/example.html"
        >
          Next
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Last page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-right"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</jest>
`;

exports[`OE - Pagination renders correctly 1`] = `
<jest>
  <nav
    aria-label="Test aria label"
    role="navigation"
  >
    <ul
      class="pagination"
    >
      <li
        class="page-item"
      >
        <a
          aria-label="First page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-left"
            />
          </svg>
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Previous page"
          class="page-link"
          href="/example.html"
        >
          Previous
        </a>
      </li>
      <li
        class="page-item disabled"
      >
        <a
          aria-disabled="true"
          class="page-link"
          href="/example.html"
          tabindex="-1"
        >
          1
        </a>
      </li>
      <li
        class="page-item active"
      >
        <a
          aria-current="page"
          class="page-link"
          href="/example.html"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="/example.html"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Next page"
          class="page-link"
          href="/example.html"
        >
          Next
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Last page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-right"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</jest>
`;

exports[`OE - Pagination renders correctly as glossary 1`] = `
<jest>
  <nav
    aria-label="Test aria label"
    role="navigation"
  >
    <ul
      class="pagination pagination--glossary"
    >
      <li
        class="page-item"
      >
        <a
          aria-label="First page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-left"
            />
          </svg>
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Previous page"
          class="page-link"
          href="/example.html"
        >
          Previous
        </a>
      </li>
      <li
        class="page-item disabled"
      >
        <a
          aria-disabled="true"
          class="page-link"
          href="/example.html"
          tabindex="-1"
        >
          1
        </a>
      </li>
      <li
        class="page-item active"
      >
        <a
          aria-current="page"
          class="page-link"
          href="/example.html"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="/example.html"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Next page"
          class="page-link"
          href="/example.html"
        >
          Next
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Last page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-right"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</jest>
`;

exports[`OE - Pagination sm renders correctly 1`] = `
<jest>
  <nav
    aria-label="Test aria label"
    role="navigation"
  >
    <ul
      class="pagination pagination-sm"
    >
      <li
        class="page-item"
      >
        <a
          aria-label="First page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-left"
            />
          </svg>
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Previous page"
          class="page-link"
          href="/example.html"
        >
          Previous
        </a>
      </li>
      <li
        class="page-item disabled"
      >
        <a
          aria-disabled="true"
          class="page-link"
          href="/example.html"
          tabindex="-1"
        >
          1
        </a>
      </li>
      <li
        class="page-item active"
      >
        <a
          aria-current="page"
          class="page-link"
          href="/example.html"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="/example.html"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Next page"
          class="page-link"
          href="/example.html"
        >
          Next
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          aria-label="Last page"
          class="page-link"
          href="/example.html"
        >
          <svg
            class="bi icon--fluid"
          >
            <use
              xlink:href="/icons.svg#chevron-double-right"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</jest>
`;
